am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -123,9 +123,9 @@ public void grantExplicitResourceAccess(@NotNull @Valid ExplicitAccessGrant acce
      * @throws PersistenceException if any persistence errors were encountered
      */
     @AuditLog("explicit access revoked by '{{mdc:caller}}' to resource '{{accessMetadata.resourceId}}' defined as "
-        + "'{{accessMetadata.resourceDefinition.serviceName}}|{{accessMetadata.resourceDefinition.resourceType}}|"
-        + "{{accessMetadata.resourceDefinition.resourceName}}' from accessor '{{accessMetadata.accessorId}}' "
-        + "with relationship '{{accessMetadata.relationship}}': {{accessMetadata.attribute}}")
+        + "'{{accessMetadata.serviceName}}|{{accessMetadata.resourceType}}|{{accessMetadata.resourceName}}' "
+        + "from accessor '{{accessMetadata.accessorId}}' with relationship '{{accessMetadata.relationship}}': "
+        + "{{accessMetadata.attribute}}")
     public void revokeResourceAccess(@NotNull @Valid ExplicitAccessMetadata accessMetadata) {
         jdbi.useExtension(AccessManagementRepository.class,
```

Comment:
not able to see any test case checks optional resource name and service name either in Lib or API.

Reply:
👍 , added integration test for null resource name and service name